components/basic/MoveToTop.js

```diff
@@ -19,10 +19,15 @@ const ScrollButton = () => {
   };
 
   const scrollToTop = () => {
-    window.scrollTo({
-      top: 0,
-      behavior: "smooth",
-    });
+    const scrollDuration = 500;
+    const scrollStep = -window.scrollY / (scrollDuration / 15); 
+    const scrollInterval = setInterval(() => {
+      if (window.scrollY === 0) {
+        clearInterval(scrollInterval); 
+      } else {
+        window.scrollBy(0, scrollStep); 
+      }
+    }, 15);
   };
 
   window.addEventListener("scroll", toggleVisible);
```

Comment:
Can you optimize the functionality by removing the event listener from dom by using the useEffect? Maybe we actually don't need a custom scroll after that....

Reply:
Sure